server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -503,6 +503,20 @@
         Setting.Property.IndexScope
     );
 
+    /**
+     * Expert: sets the amount of time to wait for merges (during {@link org.apache.lucene.index.IndexWriter#commit}
+     * or {@link org.apache.lucene.index.IndexWriter#getReader(boolean, boolean)}) returned by MergePolicy.findFullFlushMerges(...).
+     * If this time is reached, we proceed with the commit based on segments merged up to that point. The merges are not
+     * aborted, and will still run to completion independent of the commit or getReader call, like natural segment merges.
+     */
+    public static final Setting<TimeValue> INDEX_MAX_FULL_FLUSH_MERGE = Setting.timeSetting(
+        "index.max_full_flush_merge",
```

Comment:
```suggestion
        "index.merge_on_flush.max_full_flush_merge_wait_time",
```

or a shorter setting name?

Reply:
:+1: long version is fine I think, more informative